.github/workflows/typecheck.yml

```diff
@@ -0,0 +1,27 @@
+on:
+  push:
+    branches:
+      - master
+      - main
+      - default
+  pull_request:
+    branches:
+      - '**'
+
+name: Typecheck
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x]
```

Comment:
any reason to stick with 12.x, can we bump to 14.x to future proof this a bit more?

Reply:
I honestly wish there was a way to stay whatever the current stable version is, but I can't find a way to do that. Chose 12.x because it's actually newer that `10.x` used here and current LTS. Happy to put whatever makes most sense.

Please note that this just runs the typescript not any of the code here.

Reply:
yeah, so I think 14.x is going to be the best bet here, it's got a much longer life than 12.x ahead of it and is 2 weeks away from being LTS so we may as well start using it in places like this. up to you though.

The labels thing has been an ongoing discussion for a while for setup-node with folks wanting it to copy the nvm style so we could do the same things that we could do with travis, alas no serious movement: https://github.com/actions/setup-node/issues/26